TextToSpeech/TextToSpeechJsonRpc.cpp

```diff
@@ -204,9 +235,18 @@ namespace Plugin {
 
         if(parameters.HasLabel("primvolduckpercent")) {
            int8_t  primVolDuck;
-           primVolDuck = (int8_t) std::stoi(GET_STR(parameters,"primvolduckpercent","-1"));
-           _tts->SetPrimaryVolDuck(primVolDuck);
+           InputValidation::Instance().addValidator("primvolduckpercent", ExpectedValues<std::string>("^-?[0-9]+$"));
```

Comment:
Is a negative value expected?

Reply:
For the Testcase: setting value less than minimum value (-1) is being set, at JSON RPC level just checking if the provided input is number and at COM RPC level checking the range. 

So, I thought checking negative numbers as well.